src/config.ts

```diff
@@ -100,30 +106,93 @@ export function coerceAddressToBytes32(value: string): string {
   throw new Error(`Invalid value ${value}, must be a 20 or 32 byte hex string`);
 }
 
-export function buildCoreConfig(
+export function buildIsmConfig(
+  owner: types.Address,
+  remotes: ChainName[],
+): RoutingIsmConfig {
+  const mergedMultisigIsmConfig: ChainMap<MultisigIsmConfig> = objMerge(
+    defaultMultisigIsmConfigs,
+    multisigIsmConfig,
+  );
+  return {
+    owner,
+    type: ModuleType.ROUTING,
+    domains: Object.fromEntries(
+      remotes.map((remote) => [remote, mergedMultisigIsmConfig[remote]]),
+    ),
+  };
+}
+
+export function buildIsmConfigMap(
   owner: types.Address,
   chains: ChainName[],
+  remotes: ChainName[],
+): ChainMap<RoutingIsmConfig> {
+  return Object.fromEntries(
+    chains.map((chain) => {
+      const ismConfig = buildIsmConfig(
+        owner,
+        remotes.filter((r) => r !== chain),
+      );
+      return [chain, ismConfig];
+    }),
+  );
+}
+
+export function buildCoreConfigMap(
+  owner: types.Address,
+  local: ChainName,
+  remotes: ChainName[],
 ): ChainMap<CoreConfig> {
   const configMap: ChainMap<CoreConfig> = {};
-  for (const local of chains) {
-    const multisigIsmConfigs: ChainMap<MultisigIsmConfig> = {};
-    const mergedMultisigIsmConfig: ChainMap<MultisigIsmConfig> = objMerge(
-      defaultMultisigIsmConfigs,
-      multisigIsmConfig,
-    );
-    for (const remote of chains) {
-      if (local === remote) continue;
-      multisigIsmConfigs[remote] = mergedMultisigIsmConfig[remote];
-    }
-    configMap[local] = {
-      owner,
-      multisigIsm: multisigIsmConfigs,
-    };
-  }
+  configMap[local] = {
+    owner,
+    defaultIsm: buildIsmConfig(owner, remotes),
+  };
   return configMap;
 }
 
-export function buildIgpConfig(
+export function buildRouterConfigMap(
+  owner: types.Address,
+  chains: ChainName[],
+  addressesMap: HyperlaneAddressesMap<any>,
+): ChainMap<RouterConfig> {
+  const routerConfigFactories = {
+    mailbox: new Mailbox__factory(),
+    defaultIsmInterchainGasPaymaster: new OverheadIgp__factory(),
+  };
+  const filteredAddressesMap = filterAddressesMap(
+    addressesMap,
+    routerConfigFactories,
+  );
+  return Object.fromEntries(
+    chains.map((chain) => {
+      const routerConfig: RouterConfig = {
```

Comment:
Def not a blocking issue but wondering if it's worth making a util function for this `Object.fromEntries`...`.map` pattern you're using in a few places.